### ISSUE TITLE
Pods are schduled to hosts that are unknown

### ISSUE THREAD

**Issue by rrati** [GoogleCloudPlatform/kubernetes#5545]
# kubectl get minions
NAME                                                   LABELS              STATUS
...
host1   <none>              Unknown

Pods are getting scheduled to host1 despite it being in an unknown state:

Mar 17 13:40:04 master kube-scheduler[7477]: I0317 13:40:04.525316    7477 factory.go:303] Attempting to bind my-hostname-density100-1d67cbee-ccab-11e4-b5ba-ecf4bbc72672qp4v to host1

This is further worsened by the controller-manager periodically deleting all pods on host1:

Mar 17 14:32:26 master kube-controller-manager[7478]: I0317 14:32:26.237110    7478 nodecontroller.go:426] Delete all pods from host1

The controller-manager seems to do this every 10 seconds.  Pods shouldn't be scheduled to a machine that can't run them.

**Comment by timothysc** [comment]
Please add to performance and reliability label.

**Comment by erictune** [comment]
labeled

**Comment by dchen1107** [comment]
Why NodePhase is Unknown? cc/ @ddysher

**Comment by rrati** [comment]
In our case the node is one that has been configured via KUBELET_ADDRESSES on the controller-manager but has not checked in or been started up.  This could happen with nodes dropping out of the cluster due to hardware/networking issues.

**Comment by timothysc** [comment]
More color, the node was in the list for a while but was only online/reachable recently due to our private provisioning issue.

**Comment by yujuhong** [comment]
This function doesn't seem to filter out nodes without conditions:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/plugin/pkg/scheduler/factory/factory.go#L233
```
			// If no condition is set, we get unknown node condition. In such cases,
			// we add nodes unconditionally.
			nodes.Items = append(nodes.Items, node)
```

**Comment by ddysher** [comment]
nodes without conditions != node condition Unknown.  (the comment is a bit misleading)   However, when printing node status, if node has no conditions, we simply print Unknown.   Can you look at etcd and see the status of the node?  If it has no status, then @yujuhong gives the right anwser; otherwise, we'll need to dig more.

**Comment by piosz** [comment]
fixed by #5725

**Comment by rrati** [comment]
@piosz Yes, that is the PR that fixes this issue.
